### PR TITLE
Adds error message for REINDEX TABLE queries on distributed partition…

### PR DESCRIPTION
**Pull request by halilozanakgul**
Relevant PG commit:
a6642b3ae060976b42830b7dc8f29ec190ab05e4

### CODE REVIEW

**Review by agedemenli — APPROVED** [review]
Thank you

**Comment by agedemenli** [comment]
Can you share the relevant PG commit in the description?

File: src/backend/distributed/commands/index.c [hunk 1]
```diff
@@ -589,6 +589,13 @@ PreprocessReindexStmt(Node *node, const char *reindexCommand,
 
 		if (isCitusRelation)
 		{
+			if (PartitionedTable(relationId))
```
Comment (onderkalaci) [inline src/backend/distributed/commands/index.c:592]: What about `PartitionTable` s? Do they suffer from the same issue? You can create indexes with `ONLY` keyword to create indexes only on the partitions

Comment (halilozanakgul) [reply src/backend/distributed/commands/index.c:592]: Added a test with a partition table. They work fine.
Also REINDEX doesn't seem to have ONLY.

**Review by onderkalaci — APPROVED** [review]

**Pull request merged**